mlir/lib/Dialect/Tensor/IR/TensorOps.cpp

```diff
@@ -4332,21 +4332,25 @@ LogicalResult PackOp::canonicalize(PackOp packOp, PatternRewriter &rewriter) {
           rewriter.create<tensor::CastOp>(loc, newSrcType, packOp.getSource());
     }
     Value dest = packOp.getDest();
-    if (destShape != packOp.getDestType().getShape()) {
+    Type originalResultType = packOp.getDestType();
```

Comment:
The `getDestType` returns a RankedTensorType, so you can replace `Type` with it.

https://github.com/llvm/llvm-project/blob/04a8bffdf7b1d6e30616561de1734373375cfef5/mlir/include/mlir/Dialect/Tensor/IR/TensorOps.td#L1833-L1836

Reply:
Oh this was commented on an outdated diff, it probably made sense then...